exercises/practice/palindrome-products/.articles/is-palindrome-check/content.md

```diff
@@ -0,0 +1,165 @@
+to extract individual digits from the least significant to the most significant. 
+We can capture them in a list, and then read them again using maths again, 
+to reconstruct the number in the reverse order. Before we look at the code, 
+let's look at an example. The number is 123. 
+
+| Step | number | n % 10 | n / 10 |    list |
+|-----:|-------:|-------:|-------:|:--------|
+|    1 |    123 |      3 |     12 | [3]     |
+|    2 |     12 |      2 |      1 | [3,2]   |
+|    3 |      1 |      1 |      0 | [3,2,1] |
+
+So now we have digits, already in reversed order `[3,2,1]`. We have to combine them into a number.
+
+| Step | number | digit | number * 10 + digit |
+|-----:|-------:|------:|:--------------------|
+|    1 |      0 |     3 | 3                   |
+|    2 |      3 |     2 | 32                  |
+|    3 |     32 |     1 | 321                 |
+
+And now the same in code:
+
+```csharp
+using System.Collections.Generic;
+
+private static bool IsPalindrome(int number) { 
+    var original = number;
+    var digits = new List<int>();
+    
+    while (number > 0) { 
+        digits.Add(number % 10);
+        number /= 10;
+    }
+
+    var reversed = 0;
+    foreach(var digit in digits) {
+        reversed = reversed * 10 + digit;
+    }
+    
+    return original == reversed;
+}
+```
+
+### The math without the list
+
+The list in the above code helps us to see what is going on. 
+The numbers are extracted, stored in the list and then collected. 
+But, this can be done on the fly, using just math and loops without the storage. 
+
+```csharp
+private static bool IsPalindrome(int number) { 
+    var original = number;
+    var reversed = 0;
+    
+    while (number > 0) { 
+        reversed = reversed * 10 + number % 10;
+        number /= 10;
+    }
+    
+    return original == reversed;
+}
+```
+
+## The extension method options
+All of the above solutions correctly check if a number is a palindrome. 
+They differ in performance, something we will look at in a moment.
+But they also differ in readability. You will have to decide which one makes
+the most sense to you. 
+
+Talking about readability of code, we have another decision to make. 
+All of the above solutions can be implemented like they are above, or 
+as an [[extension method](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods)]
+
+What does it change? The code reads differently. Instead of 
+
+```csharp
+if (IsPalindrome(number)) { /* do something */ }
+```
+
+you could have
+
+```csharp
+if (number.IsPalindrome()) { /* do something */ }
+```
+
+You decide which one fits better with your code and expectations. 
+
+
+## Performance considerations
+
+OK, let's talk performance. I have tested the four methods above using
+[BenchmarkDotNet](https://benchmarkdotnet.org/). I won't share the details
+of where I run it, on what hardware as we are interested in relative performance only. 
```

Comment:
Could you maybe include the code that was used to run the benchmarks? Like https://github.com/exercism/csharp/tree/main/exercises/practice/leap/.articles/performance/code does. The article itself can then link to the code (see https://exercism.org/tracks/csharp/exercises/leap/articles/performance), which allows students to reproduce it locally.